test/parallel/test-stream2-readable-bytesread.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+require('../common');
+var assert = require('assert');
+
+var Readable = require('stream').Readable;
+var Duplex = require('stream').Duplex;
+var Transform = require('stream').Transform;
+
+(function() {
+  var readable = new Readable({
+    read: function(n) {
+      var i = this._index++;
+      if (i > this._max)
+        this.push(null);
+      else {
+        this.push(new Buffer('a'));
```

Comment:
@trevnorris seriously, let's not advertise it as a new default. I don't want to constrain our APIs with this recommendations.